gateway/ln-gateway/src/lib.rs

```diff
@@ -1545,6 +1560,236 @@ pub(crate) async fn fetch_lightning_node_info(
     Ok((node_pub_key, alias, network))
 }
 
+impl Gateway {
+    async fn public_key(&self, federation_id: &FederationId) -> Option<PublicKey> {
+        self.clients.read().await.get(federation_id).map(|client| {
+            client
+                .value()
+                .get_first_module::<GatewayClientModuleV2>()
+                .keypair
+                .public_key()
+        })
+    }
+
+    pub async fn payment_info(&self, federation_id: &FederationId) -> Option<PaymentInfo> {
+        Some(PaymentInfo {
+            public_key: self.public_key(federation_id).await?,
+            payment_fees: self.payment_fees(),
+            outgoing_cltv_delta: OUTGOING_CLTV_DELTA_NG,
+        })
+    }
+
+    pub fn payment_fees(&self) -> PaymentFees {
+        PaymentFees {
+            // we take a fee of one percent for outgoing contracts
+            send: PaymentFee::default(),
+            // we take a fee of one percent for incoming contracts
+            receive: PaymentFee::default(),
+        }
+    }
+
+    async fn send_payment(
```

Comment:
Should be named differently to avoid confusion. We won't be using LNv2 by default for a while due to upgradability limitations.